p2p/host/basic/basic_host.go

```diff
@@ -469,26 +469,41 @@ func (h *BasicHost) Addrs() []ma.Multiaddr {
 	return h.addrs(h.AllAddrs())
 }
 
+// mergeAddrs merges input address lists, leave only unique addresses
+func mergeAddrs(addrLists ...[]ma.Multiaddr) (uniqueAddrs []ma.Multiaddr) {
+	exists := make(map[string]bool)
```

Comment:
It would be nice to both preallocate this and `uniqueAddrs` as we know *approximately* the number of addresses we'll get. It's probably not *that* much of a performance issue but still best practices (IIRC, we'll run this function every time we connect to a new peer).

Reply:
I did a small performance test https://gist.github.com/cannium/dab43ef7b2fc7fb272b7ca583403920a, and found the version without pre-allocation is faster. So I'll leave it as is.